AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Storage/StorageEngine.swift

```diff
@@ -105,27 +109,28 @@ final class StorageEngine: StorageEngineBehavior {
 
         try storageAdapter.setUp(modelSchemas: StorageEngine.systemModelSchemas)
         if #available(iOS 13.0, *) {
-            let syncEngine = isSyncEnabled ? try? RemoteSyncEngine(storageAdapter: storageAdapter,
```

Comment:
Dont we need syncEngine when we initialize the StorageEngine for the first time?

Reply:
We don't need the syncEngine when we initialize the StorageEngine. I seperated the initialization of StorageEngine and SyncEngine. We can stop the syncEngine and reinitalized it again later.